pkg/odo/genericclioptions/context.go

```diff
@@ -12,11 +12,20 @@ import (
 	"github.com/openshift/odo/pkg/config"
 	"github.com/openshift/odo/pkg/log"
 	"github.com/openshift/odo/pkg/occlient"
-	"github.com/openshift/odo/pkg/odo/util"
+	odoutil "github.com/openshift/odo/pkg/odo/util"
 	"github.com/openshift/odo/pkg/project"
-	pkgUtil "github.com/openshift/odo/pkg/util"
+	"github.com/openshift/odo/pkg/util"
 )
 
+// ComponentOptions are generic options that are used between commands such as watch and push
+type ComponentOptions struct {
```

Comment:
I don't see it used anywhere in this PR in the git diff view shown here
And why have Sourcetype, SourcePath etc here ?
Is there any more processing to these than what can be fetched from LocalConfig?